marketplaceItems/keyboardBlueprint/entityClientScripts/keyboardKey.js

```diff
@@ -0,0 +1,155 @@
+//
+//  keyboardKey.js
+//
+//  created by Rebecca Stankus on 01/16/18
+//  updated on 01/02/19
+//  Copyright 2018 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* global Audio, Entities, Script, SoundCache */
+
+(function() { 
+    var _this;
+
+    var COLOR_CHANGE_TIMEOUT_MS = 250;
+    var AUDIO_VOLUME_LEVEL = 1;
+    var WHITE = { blue: 255, green: 255, red: 255 };
+    var BLACK = { blue: 0, green: 0, red: 0 };
+    var RANDOM_COLOR_LIGHT_1 = { blue: 162, green: 77, red: 214 };
+    var RANDOM_COLOR_LIGHT_2 = { blue: 214, green: 105, red: 77 };
+    var RANDOM_COLOR_LIGHT_3 = { blue: 87, green: 214, red: 71 };
+    var RANDOM_COLOR_LIGHT_4 = { blue: 64, green: 170, red: 219 };
+    var RANDOM_COLOR_DARK_1 = { blue: 48, green: 2, red: 77 };
+    var RANDOM_COLOR_DARK_2 = { blue: 102, green: 37, red: 13 };
+    var RANDOM_COLOR_DARK_3 = { blue: 11, green: 54, red: 5 };
+    var RANDOM_COLOR_DARK_4 = { blue: 0, green: 64, red: 94 };
+    var KEY_COLOR_INDEX_START = 13;
+    var KEY_COLOR_LENGTH = 5;
+    var KEY_NUMBER_INDEX = 19;
+    var MINIMUM_KEY_REPEAT_MS = 10;
+
+    var sound;
+    var keyDefaultColor;
+    var keyPosition;
+    var playing = false;
+
+    var Key = function() {
+        _this = this;
+    };
+
+    Key.prototype = {
+
+        /* ON PRELOAD: Save a reference to this key and set up some data about its position, sound, and color */
+        preload: function(entityID){
+            _this.entityID = entityID;
+            _this.setKeyData();
+        },
+
+        /* ON COLLISION WITH AN ENTITY: When an entity starts to collide with this keyboard key, change the key color, 
+        and play the corresponding key sound. When the entity stops colliding, return the key to its default color */
+        collisionWithEntity: function(thisEntity, otherEntity, collision) {
```

Comment:
Don't put too much time into rearchitecture - please focus on the other comments first! If you have the time, consider this comment.

Reply:
I will leave this for now with a statement in the readme as follows: "Known Issue: If two people press a keyboard key at once, it will return to default color after the first user releases the key. The second user will still be pressing the key but the key will not be highlighted to show that it is being pressed. Upon the second user releasing the key, it will respond as expected again"